nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/util/JarParser.java

```diff
@@ -61,80 +69,92 @@ public JarParser() throws Exception {
     this.builder = factory.newDocumentBuilder();
   }
 
-  public Optional<P2Attributes> getAttributesFromJarFile(final JarInputStream jis) throws Exception
+  public Optional<P2Attributes> getAttributesFromManifest(final JarInputStream jis) throws InvalidMetadataException
   {
-    // First try Features XML
-    P2Attributes p2Attributes = getAttributesFromFeatureXML(jis);
-
-    if(isNull(p2Attributes)) {
-      // Second try Manifest
-      p2Attributes = getAttributesFromManifest(jis);
-    }
-
-    return ofNullable(p2Attributes);
-  }
-
-  private P2Attributes getAttributesFromFeatureXML(final JarInputStream jis) throws Exception {
+    P2Attributes p2Attributes = null;
+    JarEntry jarEntry;
     try {
-      JarEntry jarEntry;
-      while ((jarEntry = jis.getNextJarEntry()) != null) {
-        if (jarEntry.getName().equals(XML_FILE_NAME)) {
-          return getValueFromJarEntry(jis);
+      while (p2Attributes == null && (jarEntry = jis.getNextJarEntry()) != null) {
+        if (jarEntry.getName().startsWith(MANIFEST_FILE_PREFIX)) {
+
+          Manifest manifest = jis.getManifest();
+          if (manifest == null) {
+            manifest = new Manifest(jis);
+          }
+          Attributes mainManifestAttributes = manifest.getMainAttributes();
+          String name = normalizeName(mainManifestAttributes.getValue("Bundle-SymbolicName"));
```

Comment:
If we created Manifest before on line 82 so `getValue("Bundle-SymbolicName")` could return null and after that in `normalizeName` we call `split` on null object

Reply:
we create new Manifest based on JarInputStream, so it parsed when it created, and it should have "Bundle-SymbolicName". Anyway I added check for null in normalizeName method.